openraft/tests/membership/t45_remove_unreachable_follower.rs

```diff
@@ -0,0 +1,65 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Result;
+use maplit::btreeset;
+use openraft::Config;
+
+use crate::fixtures::init_default_ut_tracing;
+use crate::fixtures::RaftRouter;
+
+/// Replication should stop after a **unreachable** follower is removed from membership.
+#[async_entry::test(worker_threads = 8, init = "init_default_ut_tracing()", tracing_span = "debug")]
+async fn stop_replication_to_removed_unreachable_follower_network_failure() -> Result<()> {
```

Comment:
Need to add a case to test a node which is removed recovered after some time.

A node is network partitioned from the cluster, and removed from leader after some error happen, after that the node's network recovered, test if the node can catch up the leader's log, etc.

Reply:
The removed node should not catch up. It is removed from membership and replication is stopped.

When the network is recovered, the removed node will try to elect and will always fail.